ReactWindows/ReactNative.Shared/Views/TextInput/ReactTextBox.cs

```diff
@@ -104,6 +133,57 @@ public int IncrementEventCount()
             return Interlocked.Increment(ref _eventCount);
         }
 
+#if WINDOWS_UWP
+        protected override void OnApplyTemplate()
+#else
+        public override void OnApplyTemplate()
+#endif
+        {
+            base.OnApplyTemplate();
+
+            if (_deleteButton != null)
+            {
+                if (DeleteButtonVisibilityToken.HasValue)
+                {
+#if WINDOWS_UWP
+                    _deleteButton.UnregisterPropertyChangedCallback(Button.VisibilityProperty, (long)DeleteButtonVisibilityToken);
+#endif
+                }
+            }
+
+            _deleteButton = (Button)GetTemplateChild("DeleteButton");
+#if WINDOWS_UWP
+            DeleteButtonVisibilityToken = _deleteButton.RegisterPropertyChangedCallback(Button.VisibilityProperty, (DependencyObject d, DependencyProperty dp) => UpdateDeleteButtonVisibility());
+#endif
+            TextChanged += OnTextChanged;
```

Comment:
The usual place to register for events like this is OnLoad - that is to avoid issues with double registration in case OnApplyTemplate is called multiple times